### ISSUE TITLE
No reproducible results when using Tensorflow backend

### ISSUE THREAD

**Issue by ivan-marroquin** [keras-team/keras#10351]
Hi everybody,

I have a windows laptop with the following configuration:
windows 7
Quadro M3000M video card
cuda 9.0 and its respective cuDNN library
conda 5 with Tensorflow-gpu 1.8.0 and Keras 2.1.6

I have a simple MLP to train using Hyperopt package to find the best hyperparameters configuration. 

To get reproducible results, I fixed the seed with the command: 
set_random_seed(1969)

Unfortunately, I get the different accuracy results and hyperparameters values at each run.

Any suggestions?
Many thanks,

Ivan

Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [x ] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps

- [x ] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

A copy of the script and input files can be found in the attached zip file

[NoReproducible_Results_Tensorflow_backend.zip](https://github.com/keras-team/keras/files/2068924/NoReproducible_Results_Tensorflow_backend.zip)

**Comment by deeiip** [comment]
May be you can have a look at #4875

**Comment by ivan-marroquin** [comment]
Thanks for the suggestion. 

In my case, I am not saving/reloading the neural network. I just display the results on a DOS console once the training face finishes.

Do you have other suggestions? I also reported this incident #10337 . It seems that there is something "weird" in Keras 2.1.6

Any other suggestion?
